integration/querydsl/blaze-persistence-querydsl-testsuite/src/test/java/com/blazebit/persistence/querydsl/BasicQueryTest.java

```diff
@@ -0,0 +1,744 @@
+        doInJPA(entityManager -> {
+            Map<Long, String> blep = new BlazeJPAQuery<Document>(entityManager, cbf).from(document)
+                    .where(document.name.length().gt(1))
+                    .groupBy(document.id, document.name)
+                    .transform(GroupBy.groupBy(document.id).as(document.name));
+
+            document.getRoot();
+            assertFalse(blep.isEmpty());
+        });
+    }
+
+    @Test
+    public void testAssociationJoin() {
+        doInJPA(entityManager -> {
+            Map<Person, List<Document>> booksByAuthor = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .from(person)
+                    .innerJoin(person.ownedDocuments, document)
+                    .transform(GroupBy.groupBy(person).as(GroupBy.list(document)));
+
+            assertNotNull(booksByAuthor);
+        });
+    }
+
+    // NOTE: Entity joins are only supported on Hibernate 5.1+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class, NoHibernate42.class, NoHibernate43.class, NoHibernate50.class })
+    public void testEntityJoin() {
+        Assume.assumeTrue(jpaProvider.supportsEntityJoin());
+
+        doInJPA(entityManager -> {
+            QPerson otherAuthor = new QPerson("otherAuthor");
+            QDocument otherBook = new QDocument("otherBook");
+            Map<Person, List<Document>> booksByAuthor = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .from(otherAuthor)
+                    .innerJoin(otherBook).on(otherBook.owner.eq(otherAuthor))
+                    .transform(GroupBy.groupBy(otherAuthor).as(GroupBy.list(otherBook)));
+
+            assertNotNull(booksByAuthor);
+        });
+    }
+
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class })
+    public void testFromValues() {
+        doInJPA(entityManager -> {
+            Document theBook = new Document();
+            theBook.setId(1337L);
+            theBook.setName("test");
+
+            List<Document> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .fromValues(document, Collections.singleton(theBook))
+                    .select(document)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class })
+    public void testFromValuesAttributes() {
+        doInJPA(entityManager -> {
+            StringPath bookName = Expressions.stringPath("bookName");
+
+            List<String> fetch = new BlazeJPAQuery<Document>(entityManager, cbf)
+                    .fromValues(document.name, bookName, Collections.singleton("book"))
+                    .select(bookName)
+                    .fetch();
+
+            assertNotNull(fetch);
+        });
+    }
+
+    @Test
+    @Category({ NoMySQL.class, NoFirebird.class, NoDatanucleus.class, NoEclipselink.class, NoOpenJPA.class })
```

Comment:
Add a comment like e.g. `// NOTE: No advanced sql support for datanucleus, eclipselink and openjpa yet` and also `// NOTE: MySQL only supports the UNION set operation`